test/e2e/metrics_test.go

```diff
@@ -88,12 +88,14 @@ func testPrometheus(t *testing.T) {
 		metrics.WindowsMetricsResource, metav1.GetOptions{})
 	require.NoError(t, err)
 
-	if gc.numberOfNodes == 0 {
+	if len(gc.allNodes()) == 0 {
 		// check if all entries in subset are deleted when there are no Windows Nodes
-		require.Equal(t, 0, len(windowsEndpoints.Subsets))
+		// TODO: BYOH cleanup isn't done yet so only check that Nodes from Machines are properly removed
```

Comment:
Where are we deleting the BYOH Machine then?

Reply:
So won't this end up leaking BYOH Machines in CI?

Reply:
Just to clarify, I want to ensure that we are deleting the `MachineSet` created without the Windows label. Just because BYOH cleanup is not implemented does not mean we can keep leaking Machines until then. Or am I missing the spot where this is getting deleted.

Reply:
Sorry, I misunderstood your first comment. By cleanup I mean cleanup that occurs when a BYOH Node is deleted, in this case the endpoints entries for the BYOH nodes arent being removed. The actual Machineset is being deleted along with the Machine controller MachineSet within the deletion suite.

Reply:
Changing the comment to be more clear
`cleanup -> metrics cleanup`